lib/update/cluster/phases/openebs.go

```diff
@@ -0,0 +1,315 @@
+	jobName := k8sutil.MakeJobName(k8sJobPrefix, p.Volume)
+	out, err := k8sutil.ExecJob(ctx, jobName, k8sNamespace, volumeUpgradeJobTemplate, &VolumeUpgrade{Volume: p.Volume,
+		FromVersion: p.FromVersion, ToVersion: p.ToVersion, JobName: jobName}, p.Client)
+
+	p.Infof("OpenEBS volume upgrade job output: %v", out)
+
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	return nil
+}
+
+// Rollback gets executed when a rollback is requested
+func (p *PhaseUpgradeVolume) Rollback(context.Context) error {
+	p.Warnf(rollbackNotSupported(), "volume", p.Volume, p.FromVersion, p.ToVersion)
+
+	return nil
+}
+
+// PreCheck gets executed before the upgrade steps
+func (*PhaseUpgradeVolume) PreCheck(ctx context.Context) error {
+	return nil
+}
+
+// PostCheck gets executed after the upgrade steps
+func (*PhaseUpgradeVolume) PostCheck(context.Context) error {
+	return nil
+}
+
+// The upgrade jobs are taken from the following OpenEBS upgrade procedure:
+// https://github.com/openebs/openebs/blob/master/k8s/upgrades/README.md
+var poolUpgradeJobTemplate = template.Must(template.New("upgradePool").Parse(`
+#This is an example YAML for upgrading cstor SPC.
+#Some of the values below needs to be changed to
+#match your openebs installation. The fields are
+#indicated with VERIFY
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  #VERIFY that you have provided a unique name for this upgrade job.
+  #The name can be any valid K8s string for name. 
+  name: {{.JobName}}
+
+  #VERIFY the value of namespace is same as the namespace where openebs components
+  # are installed. You can verify using the command:
+  # kubectl get pods -n <openebs-namespace> -l openebs.io/component-name=maya-apiserver
+  # The above command should return status of the openebs-apiserver.
+  namespace: openebs
+spec:
+  template:
+    spec:
+      #VERIFY the value of serviceAccountName is pointing to service account
+      # created within openebs namespace. Use the non-default account.
+      # by running kubectl get sa -n <openebs-namespace>
+      serviceAccountName: openebs-maya-operator
+      containers:
+      - name:  upgrade
+        args:
+        - "cstor-spc"
+
+        # --from-version is the current version of the pool
+        - "--from-version={{.FromVersion}}"
+
+        # --to-version is the version desired upgrade version
+        - "--to-version={{.ToVersion}}"
+
+        # Bulk upgrade is supported
+        # To make use of it, please provide the list of SPCs
+        # as mentioned below
+        - "{{.Pool}}"
+
+        #Following are optional parameters
+        #Log Level
+        - "--v=4"
+        #DO NOT CHANGE BELOW PARAMETERS
+        env:
+        - name: OPENEBS_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        tty: true
+
+        # the image version should be same as the --to-version mentioned above
+        # in the args of the job
+        image: openebs/m-upgrade:{{.ToVersion}}
```

Comment:
This looks like it's a reference to the dockerhub project, and if I follow the logic I don't see this get replaced anywhere. You'll need to include a reference to the container image within the storage-app resources, so they get vendored to the local registry, and update the path to point to the registry.